src/Codegen/Codegen.hs

```diff
@@ -141,9 +141,14 @@ getsTCon f i = do
 
 -- | Lookup some information associated with a data constructor.
 getsDCon :: (DConInfo -> a) -> I.DConId -> GenFn a
-getsDCon f i = do
-  Just a <- fmap f . (`dconInfo` i) <$> gets fnTypeInfo
-  return a
+getsDCon f i = 
+  case i of
+    I.DConId "(,)" -> do 
+                   Just a <- fmap f . (`dconInfo` I.DConId "Pair") <$> gets fnTypeInfo
+                   return a
+    _ -> do 
+         Just a <- fmap f . (`dconInfo` i) <$> gets fnTypeInfo
+         return a
```

Comment:
1) Run the formatter over this code
2) Instead of a case statement, use pattern matching by inserting between lines 143 and 144:
`getsDCon f (I.DConId "(,)") = do ...`
3) after you're done with the edits, run the formatter again :)